deploy.go

```diff
@@ -243,6 +242,11 @@ func (p *deploycmd) deployFunc(c *cli.Context, appName, baseDir, funcfilePath st
 		}
 	}
 
+	err = storeFuncfile(funcfilePath, funcfile)
```

Comment:
Should we be modifying the func file on deploy? Will this cause unintended changes?

Reply:
At this moment, we modify func file every time we do deploy because we bump version no matter of if deploy was successful or not. So the behaviour should be different, we define new version, run deploy with that, if the deploy was successful (images was built correctly, and pushed as well) we update funcfile struct with a new version and then write it back to func file.

That is the only flow i see here which is correct. Despite that, i tested that on the different type of use cases like the bad registry, bad build, good build with the bad registry, good build with the good registry.